modules/templates/util.go

```diff
@@ -45,3 +49,74 @@ func dict(args ...any) (map[string]any, error) {
 	}
 	return m, nil
 }
+
+func dumpVarMarshalable(v any, dumped map[uintptr]bool) (ret any, ok bool) {
+	if v == nil {
+		return nil, true
+	}
+	e := reflect.ValueOf(v)
+	for e.Kind() == reflect.Pointer {
+		e = e.Elem()
+	}
+	if e.CanAddr() {
+		addr := e.UnsafeAddr()
+		if dumped[addr] {
+			return "[dumped]", false
```

Comment:
Hmm… I'm torn about this return value.
While I agree it's better than recursively writing the same objects again, it fails in the case we have multiple objects of the same type.
Assume for example the following:
```json5
{
users:
  [{id:1, name:"first"}, {id:2, name:"second"}],
signedInUser: "[dumped]"
}
```
-> is the signed in user now first, or second?

Reply:
can we perhaps store a unique identifier for all complex objects, i.e. its address, and display it as well?

Reply:
Nope, your case won't happen.

The `[dumped]` is only for preventing from self-reference, which caused infinite recursion.

The case is `Context`: `ctx.Data["Context"] = ctx`

Reply:
Yep, you're right.
I overlooked the `defer delete`.